src/WebJobs.Script.WebHost/Controllers/KeysController.cs

```diff
@@ -18,11 +21,13 @@ public class KeysController : ApiController
 
         private readonly WebScriptHostManager _scriptHostManager;
         private readonly ISecretManager _secretManager;
+        private readonly TraceWriter _traceWriter;
 
-        public KeysController(WebScriptHostManager scriptHostManager, ISecretManager secretManager)
+        public KeysController(WebScriptHostManager scriptHostManager, ISecretManager secretManager, TraceWriter traceWriter)
         {
             _scriptHostManager = scriptHostManager;
             _secretManager = secretManager;
+            _traceWriter = traceWriter.WithPrefix($"{ScriptConstants.TracePrefixSecretManagement} API");
```

Comment:
This is what the "Source" property of TraceEvent is intended for. If you look at some core SDK logs, you'll see that they poplulate Source with the logical component, allowing log filtering, etc.